src/parser.rs

```diff
@@ -1,11 +1,14 @@
 use std;
 use nom::*;
+use chrono::{FixedOffset, DateTime};
 
 
 #[derive(Debug, Eq, PartialEq)]
 pub struct Patch<'a> {
     pub old: String,
     pub new: String,
+    pub old_timestamp: Option<DateTime<FixedOffset>>,
+    pub new_timestamp: Option<DateTime<FixedOffset>>,
```

Comment:
I'm 100% ok with breaking APIs here for now:
- `old` and `new` should become `old_filename` and `new_filename` for consistency
- OR, we could have a `file` struct, that holds the filename and timestamp, so we'd have `patch.old.filename` and `patch.old.timestamp` instead

Another question I have is, if we're capturing stuff after the filenames, how lossy should we be? GVR apparently studied a bunch of patch-producing tools, and commented in [this thread](http://www.artima.com/forums/flat.jsp?forum=106&thread=164293):

> A colleague just thanked me for writing this, and added:
>
> "Just fyi, there seems to be some small variations in the header ---/+++ lines. SVN replaces the timestamps with repository paths and revision info which invalidates Aaron's removed/created logic."

We could do something like

```rust
enum FileMeta {
    DateTime(DateTime<FixedOffset>),
    Other(&[u8]),
}
```

and then use `Option<FileMeta>` on the `Patch` struct instead. What do you think?